packages/extension/src/companion/App.tsx

```diff
@@ -0,0 +1,251 @@
+            icon={<CommentIcon />}
+          />
+        </SimpleTooltip>
+        <SimpleTooltip placement="left" content="Bookmark" appendTo="parent">
+          <Button
+            buttonSize="medium"
+            pressed={post?.bookmarked}
+            className="btn-tertiary-bun"
+            onClick={toggleBookmark}
+            icon={<BookmarkIcon />}
+          />
+        </SimpleTooltip>
+        <SimpleTooltip placement="left" content="Options" appendTo="parent">
+          <Button
+            buttonSize="medium"
+            className="btn-tertiary"
+            icon={<MenuIcon />}
+            onClick={onContextOptions}
+          />
+        </SimpleTooltip>
+        <CompanionContextMenu
+          onMessage={onMessage}
+          postData={postData}
+          onReport={report}
+          onBlockSource={blockSource}
+          onDisableCompanion={optOut}
+        />
+      </div>
+      <div className="flex flex-col p-6 h-auto rounded-l-16 border w-[22.5rem] border-theme-label-tertiary bg-theme-bg-primary">
+        <div className="flex flex-row gap-3 items-center">
+          <a href={post?.commentsPermalink} target="_parent">
+            <LogoIcon className="w-8 rounded-8" />
+          </a>
+          {post?.trending && <HotLabel />}
+        </div>
+        <p className="flex-1 my-4 typo-body">
+          <TLDRText>TLDR -</TLDRText>
+          {post?.summary ??
+            `Oops, edge case alert! Our AI-powered TLDR engine couldn't generate a summary for this article. Anyway, we thought it would be an excellent reminder for us all to strive for progress over perfection! Be relentless about learning, growing, and improving. Sometimes shipping an imperfect feature is better than not shipping at all. Enjoy the article!`}
+        </p>
+        <div
+          className="flex gap-x-4 items-center text-theme-label-tertiary typo-callout"
+          data-testid="statsBar"
+        >
+          {post?.numUpvotes > 0 && (
+            <a
+              href={post?.commentsPermalink}
+              className="flex flex-row items-center hover:underline focus:underline cursor-pointer typo-callout"
+            >
+              {post?.numUpvotes} Upvote{post?.numUpvotes > 1 ? 's' : ''}
+            </a>
+          )}
+          {post?.numComments > 0 && (
+            <span>
+              {post?.numComments.toLocaleString()}
+              {` Comment${post?.numComments === 1 ? '' : 's'}`}
+            </span>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default function App({
+  postData,
+  settings,
+}: {
+  postData: BootData;
+  settings;
+}): ReactElement {
+  const [isOptOutCompanion, setIsOptOutCompanion] = useState<boolean>(
+    settings?.optOutCompanion,
+  );
+  if (isOptOutCompanion) {
+    return <></>;
+  }
+
+  return (
+    <div>
+      <style>
+        @import &quot;chrome-extension://{browser.runtime.id}
+        /css/companion.css&quot;;
+      </style>
```

Comment:
Is there no other way to achieve it? How do we do it in the extension?

Reply:
The main issue is actually shadow dom not being able to access regular imports in React.
I can have another look at it though.

Reply:
Actually, I'm thinking of a webpack trick perhaps. The best way would be to inline the CSS to there during the build. Not sure if it's easy to achieve though.

Reply:
I tried some of the known webpack tips:
- import directly -> doesn't build in the shadow dom
- import as variable and inject output -> outputs empty

What would be the benefit of doing that vs the import like this?
As that companion.css gets dynamically build for every build. (It's a packed version of the global.css)

Reply:
To streamline the way we use global css files but don't put any more time into it. It's not that important